### ISSUE TITLE
No rollbacks in session.executeLambda

### ISSUE THREAD

**Issue by KotoJallow** [awslabs/amazon-qldb-driver-nodejs#11]
Transactional base executions via session.executeLambda does not `rollback` when an error occurs within the transaction. I wrote a wrong query within a transaction and got the BadRequestException error that's fine but mutations within that transaction were successfully executed. No rollbacks.

**Comment by othonrsanchez** [comment]
Hi @KotoJallow , thanks for reaching out.

We want to know more about the issue. Can you please share details?

1. Sample code that you are using to execute the transaction?
1. Output logs 

Thanks

**Comment by KotoJallow** [comment]
@othonrsanchez Thank you, I solved it. From my observance, the issue was due to a handled error from the async call. I tested it over and over again, if I provided
E.g
const alhm = async (txn,query) => txn.executeInline(query).catch((err) => console.log(err));
later call this function in session.executeLambda, the error is caught and `no rollbacks'
const alhm = async (txn,query) => txn.executeInline(query) without .catch()
execute session.executeLambda with a try catch, the error resulting from 'alhm' is handled in the catc
and the transaction rollbacks.
Yea it works....
